base/server/cmscore/src/com/netscape/cmscore/ldapconn/PKISocketFactory.java

```diff
@@ -162,23 +162,28 @@ public Socket makeSocket(String host, int port) throws LDAPException {
             s.setKeepAlive(keepAlive);
 
         } catch (Exception e) {
-            // for auditing
-            String localIP = "localhost";
-            try {
-                localIP = InetAddress.getLocalHost().getHostAddress();
-            } catch (UnknownHostException e2) {
-                // default to "localhost";
-            }
-            SignedAuditEvent auditEvent;
-            auditEvent = ClientAccessSessionEstablishEvent.createFailureEvent(
+            if (!external) {
+                // for auditing
+                String localIP = "localhost";
+                try {
+                    localIP = InetAddress.getLocalHost().getHostAddress();
+                } catch (UnknownHostException e2) {
+                    // default to "localhost";
+                }
+
+                SignedAuditEvent auditEvent;
+                auditEvent = ClientAccessSessionEstablishEvent.createFailureEvent(
                         localIP,
                         host,
                         Integer.toString(port),
                         "SYSTEM",
                         "connect:" +e.toString());
-            signedAuditLogger.log(auditEvent);
+                signedAuditLogger.log(auditEvent);
+                CMS.debug(e);
+            } else {
+                log(Level.SEVERE, null, e);
```

Comment:
Lines 182-184 could moved outside of `!external` and merged into a single ine:
```
log(Level.SEVERE, null, e);
```

Reply:
for 10_5, I don't think "log(Level.SEVERE, null, e);" goes into the same debug log file as CMS.debug(e), isn't it?